lib/service-manager.js

```diff
@@ -202,23 +202,24 @@ prototype.onApiRequest = function(req, callback) {
 // XXX(sam) this includes updates made from PM, so setStartOptions() may
 // need to filter out calls if there is no change.
 prototype.onInstanceUpdate = function(instance, callback) {
-  // XXX(rmg) there's a bug in here somewhere.. setImmediate breaks things
-  return process.nextTick(callback);
-  /* FIXME set-size is broken ATM
   var self = this;
-  var cpus = 'STRONGLOOP_CLUSTER' in process.env ? // XXX should be service.env?
-    process.env.STRONGLOOP_CLUSTER : 'CPU';
+  var CPUS = 'STRONGLOOP_CLUSTER' in process.env ?
```

Comment:
This should be coming from the models/config somewhere, shouldn't it? Otherwise we're overriding the app settings with whatever the PM's environment is set to, aren't we?